apparmor-utils: fix libapparmor.python cross-build

Princemachiavelli opened the pull request:
The `libapparmor.python` output fails to cross-build due to the supplied python environment (just containing `setuptools`) breaking the normal splicing of buildInputs and nativeBuildInputs. Additionally `python-check` needs to provide values and paths for targetPlatform while executable on the hostPlatform; luckily this can be achieved by using the targetPlatfrom's `python-check` instead of the `makeCWrapped` binary.

I'm not sure why but the shared library built by `setuptools` retains the platform name of the buildPlatform despite being the correct output for the targetPlatform. I'm just doing a simple rename to fix this but please let me know if anyone knows a more correct solution.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
  
  ## Testing
   - [x] legacyPackages.x86_64-linux.nixosTests.apparmor
   - [ ] legacyPackages.aarch64-linux.nixosTests.apparmor
   - [ ] legacyPackages.x86_64-linux.pkgsCross.aarch64-multiplatform.nixosTests.apparmor

Artturin commented:
All the binaries in the output seem to use python, and I assume they use `libapparmor.python` so it seems that it should be kept marked broken?

Princemachiavelli committed:
apparmor-utils: fix libapparmor.python cross-build

Use `python-config` that is executable on the build platform
while returns correct values for the targetPlatform.

Fix incorrect python shared library name from output of setuptools
build_ext.

Princemachiavelli committed:
tests(apparmor): fix cross-building test

FliegendeWurst reviewed (changes requested):
Please rebase onto staging and make sure your commit messages follow CONTRIBUTING.md.

Are you sure the Python fixes are correct? (see Artturin's comment)
